File: backend/src/main/java/com/cq/maintenance/equipment/entity/EquipmentStatus.java
package com.cq.maintenance.equipment.entity;

import com.baomidou.mybatisplus.annotation.EnumValue;
import com.fasterxml.jackson.annotation.JsonValue;

public enum EquipmentStatus {
    PENDING("待启用"), RUNNING("运行"), FAULT("故障"), REPAIRING("维修中"), STOPPED("停用"), SCRAPPED("报废");
    @EnumValue private final String code;
    private final String label;
    EquipmentStatus(String label){this.code=name();this.label=label;}
    @JsonValue public String code(){return code;}
    public String label(){return label;}
}

File: frontend/src/utils/equipmentStatus.js
export const equipmentStatuses = Object.freeze({
  PENDING: { label: '待启用', type: 'info', order: 1 },
  RUNNING: { label: '运行中', type: 'success', order: 2 },
  FAULT: { label: '故障', type: 'danger', order: 3 },
  REPAIRING: { label: '维修中', type: 'warning', order: 4 },
  STOPPED: { label: '已停用', type: 'info', order: 5 },
  SCRAPPED: { label: '已报废', type: 'danger', order: 6 },
})
export const statusLabel = (code) => equipmentStatuses[code]?.label || code
export const statusTagType = (code) => equipmentStatuses[code]?.type || 'info'
export const manualStatusActions = (code) => ({
  PENDING: [{ code: 'RUNNING', label: '启用' }],
  RUNNING: [{ code: 'STOPPED', label: '停用' }],
  STOPPED: [{ code: 'RUNNING', label: '恢复运行' }],
}[code] || [])
